### ISSUE TITLE
 Request: git Review Changes window: default layout with commit message window at bottom instead of to the right

### ISSUE THREAD

**Issue by mskyttner** [rstudio/rstudio#8624]
Since the commit message often is shorter, it would be more natural to have it at the bottom of the git review changes window, swapping its position with the "diff window", which feels more natural to position to the right of the file list (and on most large screens would give it more length and thus show more of the changes by default).

Look at all this empty space for a tiny commit message in the screenshot below.

It would be nice to by default be able to see more of the changes made in the file instead with this "diff" file view to the right of the files, which mimics how most file browsing UIs look.

And the commit message could a shorter slice at the bottom of the screen, which would feel natural with a commit buttom at the lower right corner of the screen.

![rstudio-git-review-changes-gui](https://user-images.githubusercontent.com/1715840/102337381-afac1e80-3f92-11eb-9712-045c431be940.png)

**Comment by pkq** [comment]
I think your request makes a lot of sense, but FYI, you can resize the panes using the thin gray separator. This obviously isn't exactly what you're requesting, but it does give you more control over how much of the diff window you see.

**Comment by mskyttner** [comment]
@pkq thanks for reconnecting, this request/wish is not primarily about window/pane resizing which works well as you pointed out, but more about default layout arrangement of the panes in the screen. 

This important popup window lacks what the main editor has where one can also rearrange locations of panes (starting off with a sensible initial default layout arrangement, and with a possibility to save the setting). Perhaps that is not so important here (or if the review changes had been an optional breakout (like the viewer pane), if the default layout arrangement of panes would be adjusted.

So this is more of a suggestion that the default layout arrangement be changed, since a lot of people doing development work have pretty wide landscape aspect screens and if the popup is opening up on top of the editor, there is likely to be more space available for viewing the contents of the diff if that pane had been located on the right. Higher "data to ink" ratio. I think on average the commit messages are recommended to be succinct, perhaps around 60-80 chars max, and would often fit well in a thinner slice at the bottom without much line breaking. 

So in short, the wish is for the default layout arrangement changing from a horizontal to a vertical main split in the screen/display (for both the "Changes" and the  "History" toggle mode).

Since it is close to xmas I'll add another thing to the wish list which is related, but this one is for "quick commits", when one doesn't want to bring up the popup for a more extensive review of a larger set of changes, and here is a mockup suggestion for that:

![quick-commit](https://user-images.githubusercontent.com/1715840/102461199-5b637600-4048-11eb-8e59-971ca4d6a492.png)

**Comment by mskyttner** [comment]
Related https://github.com/rstudio/rstudio/issues/7632

**Comment by ronblum** [comment]
@mskyttner Thank you for the suggestion and the comprehensive discussion! We'll consider it during our ongoing development of RStudio.

**Comment by mskyttner** [comment]
Still open and not stale.

**Comment by mskyttner** [comment]
Still open and not stale.

**Comment by mskyttner** [comment]
Still open and not stale.

**Comment by mskyttner** [comment]
Still open and not stale

**Comment by mskyttner** [comment]
Still open and not stale.

**Comment by mskyttner** [comment]
Still open and not stale, but maybe time passing has made this less relevant.

**Comment by mskyttner** [comment]
Still open and not stale, but maybe time passing has made this less relevant.
